tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/JmapSettingsSetMethodContract.scala

```diff
@@ -661,4 +670,347 @@ trait JmapSettingsSetMethodContract {
+           |        "notUpdated": {
+           |            "singleton": {
+           |                "type": "invalidArguments",
+           |                "description": "Cannot perform both a reset and a partial update simultaneously"
+           |            }
+           |        }
+           |    },
+           |    "c1"
+           |]""".stripMargin))
+
+  @Test
+  def updatePartialShouldUpdateNewState(server: GuiceJamesServer): Unit = {
+    val response =`given`
+      .body(
+        s"""{
+           |	"using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:settings"],
+           |	"methodCalls": [
+           |		[
+           |			"Settings/set",
+           |			{
+           |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |				"update": {
+           |					"singleton": {
+           |						"settings/key1": "value1"
+           |					}
+           |				}
+           |			}, "c1"
+           |		]
+           |	]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract()
+      .jsonPath()
+
+    val afterSettingsSetState: UuidState = server.getProbe(classOf[JmapSettingsProbe])
+      .getLatestState(BOB)
+
+    assertThat(response.getString("methodResponses[0][1].oldState")).isEqualTo(JmapSettingsStateFactory.INITIAL.serialize)
+    assertThat(response.getString("methodResponses[0][1].oldState")).isNotEqualTo("methodResponses[0][1].newState")
+    assertThat(response.getString("methodResponses[0][1].newState")).isEqualTo(afterSettingsSetState.serialize)
+  }
+
+  @Test
+  def updateShouldFailWhenEmptyBothUpdatePartialAndFullResetPatch(): Unit =
+      `given`
+      .body(
+        s"""{
+           |	"using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:settings"],
+           |	"methodCalls": [
+           |		[
+           |			"Settings/set",
+           |			{
+           |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |				"update": {
+           |					"singleton": {}
+           |				}
+           |			}, "c1"
+           |		]
+           |	]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .body("methodResponses[0]", jsonEquals(
+        s"""[
+           |    "Settings/set",
+           |    {
+           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |        "oldState": "${JmapSettingsStateFactory.INITIAL.serialize}",
+           |        "newState": "${JmapSettingsStateFactory.INITIAL.serialize}",
+           |        "notUpdated": {
+           |            "singleton": {
+           |                "type": "invalidArguments",
+           |                "description": "update request must not be empty"
+           |            }
+           |        }
```

Comment:
Just raising the question. Should we return `updated` with no actual changes or `notUpdated` with error when empty update object?

rf VacationResponseSet: https://github.com/apache/james-project/blob/master/server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/VacationResponseSetMethodContract.scala#L175

Reply:
I will update to similar to past others